arangod/Aql/IResearchViewNode.cpp

```diff
@@ -127,10 +127,27 @@ std::vector<Scorer> fromVelocyPack(aql::ExecutionPlan& plan, velocypack::Slice c
 // -----------------------------------------------------------------------------
 // --SECTION--                            helpers for IResearchViewNode::Options
 // -----------------------------------------------------------------------------
+namespace {
+  std::map<std::string, arangodb::aql::ConditionOptimization> conditionOptimizationTypeMap = {
+    {"auto", arangodb::aql::ConditionOptimization::Auto},
+    {"nodnf", arangodb::aql::ConditionOptimization::NoDNF},
+    {"noneg", arangodb::aql::ConditionOptimization::NoNegation},
+    {"none", arangodb::aql::ConditionOptimization::None}
+  };
+}
+
 
 void toVelocyPack(velocypack::Builder& builder, IResearchViewNode::Options const& options) {
   VPackObjectBuilder objectScope(&builder);
   builder.add("waitForSync", VPackValue(options.forceSync));
+  {
+    for (auto r : conditionOptimizationTypeMap) {
```

Comment:
To avoid copying the map's string keys:
```suggestion
    for (auto const& r : conditionOptimizationTypeMap) {
```